src/oauth-client.ts

```diff
@@ -25,6 +25,12 @@ export default class OauthClient {
       throw new Error('OAuth Client cannot operate within insecure contexts.');
     }
 
+    if (scopes.length === 0) {
+      console.warn(
+        'Instantiating an OauthClient without explicit scopes is deprecated and will be removed in the future.',
+      );
+    }
+
```

Comment:
`django-allauth` requires `scopes` to be explicitly passed, so I added this here. I'm not sure if this is the correct approach, or if we should try to get `allauth` to allow an empty scopes param upstream. The latter may not make sense because it appears that `allauth` is explicitly implementing OIDC, not just OAuth2.

Reply:
I'm not sure about upstream, but for this library, I'm actually find if we just make `scopes` required as a breaking change. We can release a new major version.

Reply:
To be clear, I'm not saying that it's the right decision to make `scopes` required, just that I'm ok with breaking changes in principle.